staging/src/k8s.io/apimachinery/pkg/api/resource/quantity_proto.go

```diff
@@ -268,6 +268,9 @@ func skipGenerated(data []byte) (n int, err error) {
 					return 0, err
 				}
 				iNdEx = start + next
+				if iNdEx < 0 {
+					return 0, ErrInvalidLengthGenerated
+				}
```

Comment:
do we really want to allow iNdEx to go backwards if `start` happened to be currently bigger than a negative `next`?

I haven't looked at this closely, but if `next` is < 0 (or maybe even <= 0), is *that* the core problem to detect, return an error for?

Reply:
I think that's the case, when index < 0 an exception needs to be thrown, otherwise it would lead to a crash (https://github.com/kubernetes/kubernetes/issues/116745).

This pull request is returning an error code that will terminate the program's continued execution.